### PR TITLE
feat: introduce foundry deploy script

### PR DESCRIPTION
**Pull request by 0x-r4bbit**
This introduces a foundry based deployment script and tests that ensure the deployment works as expected

## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [ ] Added natspec comments?
- [x] Ran `forge snapshot`?
- [x] Ran `pnpm lint`?
- [x] Ran `forge test`?

### CODE REVIEW

File: script/DeployBridge.s.sol [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            remoteTokenAddress,
+            MiniMeBase(payable(parentTokenAddress)),
+            parentSnapShotBlock,
+            tokenName,
+            decimals,
+            tokenSymbol,
+            transferEnabled
+        );
+        SNTPlaceHolder tokenController = new SNTPlaceHolder(payable(address(bridgeToken)));
+        bridgeToken.changeController(payable(address(tokenController)));
```
Comment (0x-r4bbit) [inline script/DeployBridge.s.sol:37]: @3esmit We need to decide here if we actually want a controller involved or not.

File: script/DeploymentConfig.s.sol [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    NetworkConfig public activeNetworkConfig;
+
+    address private deployer;
+
+    // solhint-disable-next-line var-name-mixedcase
+    address internal SNT_ADDRESS_MAINNET = 0x744d70FDBE2Ba4CF95131626614a1763DF805B9E;
+    // solhint-disable-next-line var-name-mixedcase
+    address internal SNT_ADDRESS_GOERLI = 0x3D6AFAA395C31FCd391fE3D562E75fe9E8ec7E6a;
+    // solhint-disable-next-line var-name-mixedcase
+    address internal STANDARD_BRDIGE_ADDRESS = 0x4200000000000000000000000000000000000010;
```
Comment (0x-r4bbit) [inline script/DeploymentConfig.s.sol:32]: There's a typo in this property that I need to fix

File: script/DeploymentConfig.s.sol [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    function getOptimismGoerliConfig() public view returns (NetworkConfig memory) {
+        return NetworkConfig({
+            deployer: deployer,
+            bridgeAddress: STANDARD_BRDIGE_ADDRESS,
+            remoteTokenAddress: SNT_ADDRESS_GOERLI,
+            parentTokenAddress: address(0),
+            parentSnapShotBlock: 0,
+            tokenName: "Goerli Optimism SNT",
```
Comment (0x-r4bbit) [inline script/DeploymentConfig.s.sol:72]: @3esmit let me know if on Goerli we want this to be called `Status Network Token` as well

File: script/DeploymentConfig.s.sol [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            decimals: 18,
+            tokenSymbol: "SNT",
+            transferEnabled: true
+        });
+    }
+
+    function getOptimismConfig() public view returns (NetworkConfig memory) {
+        return NetworkConfig({
+            deployer: deployer,
+            bridgeAddress: STANDARD_BRDIGE_ADDRESS,
```
Comment (0x-r4bbit) [inline script/DeploymentConfig.s.sol:82]: @3esmit I'm assuming both, goerli optimism and mainnet optimism use the same address of for the standard bridge. I've checked on either can and they are both instances of the bridge on both networks.

Let me know if this is incorrect.

File: script/DeployBridge.s.sol [hunk 5]
```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.19 <=0.9.0;
+
+import { MiniMeBase } from "@vacp2p/minime/contracts/MiniMeBase.sol";
+import { BaseScript } from "./Base.s.sol";
+import { DeploymentConfig } from "./DeploymentConfig.s.sol";
+import { OptimismMintableMiniMeToken } from "../contracts/optimism/OptimismMintableMiniMeToken.sol";
+import { SNTPlaceHolder } from "../contracts/SNTPlaceHolder.sol";
```
Comment (0x-r4bbit) [inline script/DeployBridge.s.sol:8]: @3esmit How do you feel about renaming this one? I find "placeholder" extremely misleading. It's effectively a token controller, so how about we rename it to `SNTTokenController` or something like that?

Comment (3esmit) [reply script/DeployBridge.s.sol:8]: Indeed. SNTOptimismController.sol would be better

File: script/DeployBridge.s.sol (hunk 1 above)
Comment (3esmit) [reply script/DeployBridge.s.sol:37]: I think we should.

File: script/DeploymentConfig.s.sol (hunk 3 above)
Comment (3esmit) [reply script/DeploymentConfig.s.sol:72]: Status Test Token

File: script/DeployBridge.s.sol (hunk 5 above)
Comment (0x-r4bbit) [reply script/DeployBridge.s.sol:8]: Okay cool, will do in a follow-up PR

**Comment by 0x-r4bbit** [comment]
@3esmit Updated the deployment config to use `Status Test Token` as token name for test nets (and anvil).
But I'm still using `SNT` over `STT` as symbol. Are you okay with that?

Based on my experience it's sometimes a pain that `SNT` has a different symbol on test nets.
I think we're doing everyone a favour if we call it `SNT` on goerli optimism as well.

Lemme know what you think.

File: script/DeployBridge.s.sol (hunk 5 above)
Comment (0x-r4bbit) [reply script/DeployBridge.s.sol:8]: Here's a PR that renames the contract accordingly https://github.com/logos-co/optimism-bridge-snt/pull/19

**Comment by 0x-r4bbit** [comment]
Before this gets merged, we need to confirm: https://github.com/logos-co/optimism-bridge-snt/pull/17#discussion_r1347277224

**Review by 3esmit — APPROVED** [review]

File: script/DeploymentConfig.s.sol (hunk 4 above)
Comment (3esmit) [reply script/DeploymentConfig.s.sol]: Yes, they use the same address.

**Pull request merged**